src/simulation/simulator.cpp

```diff
@@ -568,6 +589,46 @@ double Simulator::calculateVoltage(ConnectorItem * c0, ConnectorItem * c1) {
 	return volt0-volt1;
 }
 
+std::vector<double> Simulator::voltageVector(ConnectorItem * c0) {
+	int net0 = m_connector2netHash.value(c0);
+	std::cout << "calculateVoltageVector: ";
+	QString net0str = QString("v(%1)").arg(net0);
+
+	auto timeInfo = m_simulator->getVecInfo(QString("time").toStdString());
+	if (net0 != 0) {
+		std::cout << "calculateVoltageVector: ";
+	}
+	return m_simulator->getVecInfo(net0str.toStdString());
+}
+
+QString Simulator::generateSvgPath(std::vector<double> proveVector, std::vector<double> comVector, QString nameId, double v_div, double v_offset) {
+	std::cout << "VOLTAGE VALUES " << nameId.toStdString() << ": ";
+	QString svg;
+	if (!nameId.isEmpty())
+		svg += QString("<path id='%1' d='").arg(nameId);
+	else
+		svg += QString("<path d='");
+
+	double vScale = -12.5/v_div;
+	double y_0 = 50; // the center of the screen
+
+
+	for (int i = 0; std::min( proveVector.size(), comVector.size() ); i++) {
+		double voltage = proveVector[i] - comVector[i];
+		if (i == 0) {
+			svg.append("M 0 " + QString::number( (voltage + v_offset) * vScale + y_0, 'f', 3) + " ");
+		} else {
+			svg.append("L " + QString::number(i, 'f', 3) + " " + QString::number((voltage + v_offset) * vScale + y_0, 'f', 3) + " ");
+		}
+		std::cout << voltage << ' ';
+	}
+	svg += "' fill='red' stroke='black' stroke-width='10'/> \n";
+
+	std::cout << std::endl;
+	return svg;
+
```

Comment:
I think this and other methods might be better placed in items/oscilloscope.cpp or similar.
So, we'd create an Oscilloscope class, that can display probe and comVectors ... in the future it could also hold code to act on buttons and knobs. 

Is there a specific reason it has to be here?

Reply:
No, I also thought about that several times. I can create an oscilloscope item and move that code there.